src/charm.py

```diff
@@ -140,14 +140,49 @@ def _check_leader(self):
             self.logger.info("Not a leader, skipping setup")
             raise ErrorWithStatus("Waiting for leadership", WaitingStatus)
 
-    def _deploy_k8s_resources(self) -> None:
-        """Deploys K8S resources."""
+    def _check_and_report_k8s_conflict(self, error):
+        """Check for conflict and report it.
+
+        Returns:
+          True if error status code is 409 (conflict).
+          False otherwise.
+        """
+        if error.status.code == 409:
+            self.unit.status = MaintenanceStatus("Force applying K8S resources")
+            self.logger.warning(f"{str(error)}")
```

Comment:
```suggestion
            self.unit.status = MaintenanceStatus("Encountered conflicts on apply")
            self.logger.warning(f"Encountered a conflict when applying K8S resources (ApiError returned error code 409}")
```

Reply:
Using `{error}` produces very descriptive log message with exact problem. I don't think it is better to have a generic log message when a detailed information is available for free.
On `MaintenanceStatus` I will also disagree. By this point we are applying resources (action) thus "Force applying K8S resources". Just reporting that we encountered error does not describe what is actually going on. And second, "Encountered conflicts on apply" is really an error/blocked status not maintenance.

Reply:
This is sample log message produced by having `{error}`. I think, it is fantastic, as far as log messages go:
```
Apply failed with 1 conflict: conflict with "python-httpx" using rbac.authorization.k8s.io/v1: .rules
```